storage/Utils/RegionImpl.cc

```diff
@@ -107,6 +107,27 @@ namespace storage
     }
 
 
+    void
+    Region::Impl::adjust_block_size(unsigned int block_size)
+    {
+	assert_valid_block_size(block_size);
+
+	if (block_size == Impl::block_size)
+	    return;
+
+	if (start * Impl::block_size % block_size ||
+	    length * Impl::block_size % block_size)
+	{
+	    ST_THROW(InvalidBlockSize(block_size));
+	}
```

Comment:
For the Region class, raising this exception is ok.

But in my opinion, we should have some previous check (in libstorage-ng or in yast2-storage-ng) to prevent the combination of incompatible devices into a RAID, either caching this exception or doing the check beforehand. Any plan/idea about it?

Reply:
That's really a fringe case and can happen if you combine an odd-sized 0.5k disk with a 4k disk into a raid.

I'm fine with throwing a different exception if you have any suggestion. So that you don't start to replicate the logic in y2storage.

Not sure that this is still in scope of the pr, which is about fixing a bug.